middleware/jmx/variables.tf

```diff
@@ -0,0 +1,477 @@
+# Global
+
+variable "environment" {
+  description = "Infrastructure environment"
+  type        = string
+}
+
+# SignalFx module specific
+
+variable "notifications" {
+  description = "Notification recipients list for every detectors"
+  type        = list
+}
+
+variable "prefixes" {
+  description = "Prefixes list to prepend between brackets on every monitors names before environment"
+  type        = list
+  default     = []
+}
+
+variable "filter_custom_includes" {
+  description = "List of tags to include when custom filtering is used"
+  type        = list
+  default     = []
+}
+
+variable "filter_custom_excludes" {
+  description = "List of tags to exclude when custom filtering is used"
+  type        = list
+  default     = []
+}
+
+variable "detectors_disabled" {
+  description = "Disable all detectors in this module"
+  type        = bool
+  default     = false
+}
+
+# jmx detectors specific
+
+variable "jmx_memory_heap_usage_disabled_warning" {
+  type    = bool
+  default = null
+}
+
+variable "jmx_memory_heap_usage_notifications_warning" {
+  type    = list
+  default = []
+}
+
+variable "jmx_memory_heap_usage_disabled_critical" {
+  type    = bool
+  default = null
+}
+
+variable "jmx_memory_heap_usage_notifications_critical" {
+  type    = list
+  default = []
+}
+
+variable "jmx_memory_heap_used_aggregation_function" {
```

Comment:
do you really need different aggregation for max and used ? I think only one variable should be enough, same remark for every other detectors